DSharpPlus.Commands/Processors/SlashCommands/SlashCommandProcessor.cs

```diff
@@ -0,0 +1,442 @@
+    {
+        if (this._extension is null)
+        {
+            throw new InvalidOperationException("SlashCommandProcessor has not been configured.");
+        }
+
+        // Translate the command's name and description.
+        IReadOnlyDictionary<string, string> nameLocalizations = new Dictionary<string, string>();
+        IReadOnlyDictionary<string, string> descriptionLocalizations = new Dictionary<string, string>();
+        if (command.Attributes.OfType<SlashLocalizerAttribute>().FirstOrDefault() is SlashLocalizerAttribute localizerAttribute)
+        {
+            AsyncServiceScope scope = this._extension.ServiceProvider.CreateAsyncScope();
+            nameLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.name");
+            descriptionLocalizations = await localizerAttribute.LocalizeAsync(scope.ServiceProvider, $"{command.FullName}.description");
+            await scope.DisposeAsync();
+        }
+
+        // Convert the subcommands or parameters into application options
+        List<DiscordApplicationCommandOption> options = [];
+        if (command.Subcommands.Any())
+        {
+            foreach (Command subCommand in command.Subcommands)
+            {
+                options.Add(await this.ToApplicationParameterAsync(subCommand));
+            }
+        }
+        else
+        {
+            foreach (CommandParameter parameter in command.Parameters)
+            {
+                if (parameter.Attributes.OfType<ParamArrayAttribute>().Any())
+                {
+                    // Fill til 25
+                    for (int i = options.Count; i < 24; i++)
+                    {
+                        options.Add(await this.ToApplicationParameterAsync(command, parameter, i));
+                    }
+                }
+
+                options.Add(await this.ToApplicationParameterAsync(command, parameter));
+            }
+        }
+
+        // Create the top level application command.
+        return new(
+            name: nameLocalizations.TryGetValue("en-US", out string? name) ? name : command.Name,
+            description: descriptionLocalizations.TryGetValue("en-US", out string? description) ? description : command.Description,
+            options: options,
+            type: ApplicationCommandType.SlashCommand,
+            name_localizations: nameLocalizations,
+            description_localizations: descriptionLocalizations,
+            allowDMUsage: command.Attributes.Any(x => x is AllowDMUsageAttribute),
+            defaultMemberPermissions: command.Attributes.OfType<RequirePermissionsAttribute>().FirstOrDefault()?.UserPermissions ?? Permissions.None,
+            nsfw: command.Attributes.Any(x => x is RequireNsfwAttribute)
+        );
+    }
+
+    [SuppressMessage("Roslyn", "CA1859", Justification = "Incorrect warning in NET 8-rc2. TODO: Open an issue in dotnet/runtime.")]
```

Comment:
anything changed about this TODO?

Reply:
...no

Reply:
It made it into the .NET 8 release. I'm sure someone else has already created an issue for it.